base/util.go

```diff
@@ -590,27 +591,117 @@ func IsFilePathWritable(fp string) (bool, error) {
 	return true, nil
 }
 
-// Replaces sensitive data from the URL query string with ******.
+// SanitizeRequestURL will return a sanitised string of the URL by:
+// - Tagging mux path variables.
+// - Tagging query parameters.
+// - Replacing sensitive data from the URL query string with ******.
 // Have to use string replacement instead of writing directly to the Values URL object, as only the URL's raw query is mutable.
-func SanitizeRequestURL(requestURL *url.URL) string {
-	urlString := requestURL.String()
+func SanitizeRequestURL(req *http.Request) string {
+	urlString := SanitizeRequestURLQueryParams(req.URL)
+
+	if RedactSystemData || RedactMetadata || RedactUserData {
+		tagQueryParams(req, &urlString)
+		tagPathVars(req, &urlString)
+	}
+
+	return urlString
+}
+
+// redactedPathVars is a lookup map of path variables to redaction types.
+var redactedPathVars = map[string]string{
+	"docid":   "UD",
+	"attach":  "UD",
+	"name":    "UD",
+	"channel": "UD",
+
+	"db":        "MD",
+	"newdb":     "MD",
+	"ddoc":      "MD",
+	"view":      "MD",
+	"sessionid": "MD",
+}
+
+// tagPathVars will tag all redactble path variables in the urlString for the given request.
+func tagPathVars(req *http.Request, urlString *string) {
+	if urlString == nil {
+		return
+	}
+
+	str := *urlString
+	pathVars := mux.Vars(req)
+
+	for k, v := range pathVars {
+		switch redactedPathVars[k] {
+		case "UD":
+			str = strings.Replace(str, v, UD(v).Redact(), 1)
+		case "MD":
+			str = strings.Replace(str, v, MD(v).Redact(), 1)
+		case "SD":
+			str = strings.Replace(str, v, SD(v).Redact(), 1)
+		}
```

Comment:
Look at whether it's reasonable to include leading slash in the pattern matching to solve the double-redact problem for duplicated path variable values.  Think about whether that also covers the case where a path variable value matches another chunk of the endpoint (e.g. 4985/changes/_changes)